Tests/UnitTests/Core/Navigation/NavigationStateUpdatorsTests.cpp

```diff
@@ -108,6 +108,25 @@ struct IndexedSurfacesExtractor {
 
 }  // namespace Experimental
 
+class TestAxis : virtual public IAxis {
```

Comment:
Sorry maybe I missed this in the rest of the diff: do you use `IAxis` as part of the lookup mechanism? If I remember correctly I only intended this for debugging purposes and to use the direct types for the actual lookup call chain. 

Also: is the `virtual` needed here?

Reply:
Indeed, the `virtual` as is not needed - this is just a test axis, what I use from `IAxis` is that the grid returns

`auto axes = grid.axes()` and I access during filling, and during displaying some basic info from the `IAxis`,
such as `getNBins()` and `getBoundaryType`.

During runtime of a job, this is not needed.

Reply:
Removed the virtual.